### ISSUE TITLE
OverlayPanel: z-index issue

### ISSUE THREAD

**Issue by matinh** [primefaces/primefaces#7276]
The contents of an opened `<p:overlayPanel>` should be at the top (speaking in z-dimension) and cover other contents. However, if the `<p:overlayPanel>` is used inside a `<p:tabView>`, this is not the case for `<p:commandButton> components. They cover the overlayPanel!

A picture is worth a thouthands words:
![Image of Yaktocat](https://i.imgur.com/yv52N7V.png)

To me this seems like a CSS bug, but I couldn't find out, what the actual problem is. It seems, _opacity_, _position_ and _transform_ are related, but I couldn't say definitely.

**Reproducer**
Run this simple XHTML and click the "click me" text to reproduce:

```xhtml
    <h:form id="myForm">
      <p:tabView id="tabView" >
        <p:tab title="Test">
          <h:outputText id="trigger" value="click me" />         
            <p:overlayPanel for="trigger" style="background-color: #4c4c4c;">
              <h1 style="padding:1em">Some big Tooltip</h1>
            </p:overlayPanel>
        </p:tab>
      </p:tabView>

      <div><p:commandButton value="p:commandButton" /></div>
      <div><h:commandButton value="h:commandButton"/></div>
      <div><p:panel>p:panel - hello</p:panel></div>

    </h:form>
```

**Environment:**
 - PF Version: _8.0_
 - JSF + version: _e.g. Mojarra 2.3.9 (probably irrelevant)
 - Affected browsers: tested with _FireFox 87_, _Chromium 89_ and _Opera 76_

**Expected behavior**

The contents of the overlayPanel should not be covered by any other component like the <p:commandButton>. The button should be "under" the overlayPanel.

**Comment by melloware** [comment]
Already fixed in 10.0.0. Try this reproducer:
[pf-7276.zip](https://github.com/primefaces/primefaces/files/6399345/pf-7276.zip)

Run `mvn clean jetty:run` and navigate to http://localhost:8080/primefaces-test/test.xhtml

Screenshot:
![image](https://user-images.githubusercontent.com/4399574/116566879-1c3a1100-a8d5-11eb-9942-0b6a682004de.png)


its working fine

**Comment by melloware** [comment]
Unless you have a PRO account to report this for 8.0 to get it in Elite release. It has already been fixed in the Community 10.0.0 so there is nothing I an do here except close the ticket and mark it ask fixed in 10.0.0

**Comment by matinh** [comment]
Switching to PF 10 is currently no option for me.
Do you know if there was actually a fix/patch for this in PF10, or does it just work because of redesign or similar?

**Comment by melloware** [comment]
I have a feeling it works because of redesign either with this ticket: https://github.com/primefaces/primefaces/issues/6973

Or: https://github.com/primefaces/primefaces/issues/6978

**Comment by matinh** [comment]
Thanks so far. So I'm on my own :(
If I find an easy fix, I'll come back here.

**Comment by melloware** [comment]
OK cool it might be a zIndex thing where you might be able to  do it with CSS stting the overlayPanel to z-index 9999 or something.

**Comment by matinh** [comment]
It's definitely not a (simple) z-index thing. Changing z-index has no effect unless I set the z-index of the problematic button to -1!

It seems the problem is triggered by a combination of various style-attributes.
I could work around the problem when I defined the following CSS rules:

```css
    .ui-button {
        position:initial !important;
        -moz-transform:none !important;
        -o-transform:none !important;
        -webkit-transform:none !important;
        transform:none !important;
    }
```
Note, that this might have unwanted side-effects. I didn't test this extensively.

Unfortunately this change was not enough, when the button was disabled. In this case I had to additionally deactivate the opacity:

```css
    .ui-state-disabled {
        opacity: 1 !important;
    }
```

That's not what I want, but that was the only way I found to bring the button below the tooltip :(

To me this really looks like a bug in the HTML rendering engine. But then, it probably wouldn't be in all/most browsers. Anyway. I think I'll stop here for now. If somebody has any useful input, you're more than welcome!
